xbmc/addons/binary-addons/AddonInstanceHandler.h

```diff
@@ -14,48 +14,87 @@
+                        const AddonInfoPtr& addonInfo,
+                        uint32_t instanceNumber = 0,
+                        KODI_HANDLE parentInstance = nullptr,
+                        const std::string& instanceID = "");
+  virtual ~IAddonInstanceHandler();
+
+  ADDON_TYPE UsedType() const { return m_type; }
+  const std::string& InstanceID() { return m_instanceId; }
+  uint32_t InstanceNumber() const { return m_instanceNumber; }
+
+  std::string ID() const;
+  std::string Name() const;
+  std::string Author() const;
+  std::string Icon() const;
+  std::string Path() const;
+  std::string Profile() const;
+  AddonVersion Version() const;
+
+  ADDON_STATUS CreateInstance();
+  void DestroyInstance();
+  const AddonDllPtr& Addon() const { return m_addon; }
+  AddonInfoPtr GetAddonInfo() const { return m_addonInfo; }
+
+  virtual void OnPreInstall() {}
+  virtual void OnPostInstall(bool update, bool modal) {}
+  virtual void OnPreUnInstall() {}
+  virtual void OnPostUnInstall() {}
+
+protected:
+  KODI_ADDON_INSTANCE_INFO m_info;
+  KODI_ADDON_INSTANCE_STRUCT m_ifc;
+
+private:
+  std::shared_ptr<CSetting> GetSetting(const std::string& setting);
+
+  static bool is_instance_setting_using_default(const KODI_ADDON_INSTANCE_BACKEND_HDL hdl,
+                                                const char* id);
+  static bool get_instance_setting_bool(const KODI_ADDON_INSTANCE_BACKEND_HDL hdl,
+                                        const char* id,
+                                        bool* value);
+  static bool get_instance_setting_int(const KODI_ADDON_INSTANCE_BACKEND_HDL hdl,
+                                       const char* id,
+                                       int* value);
+  static bool get_instance_setting_float(const KODI_ADDON_INSTANCE_BACKEND_HDL hdl,
+                                         const char* id,
+                                         float* value);
+  static bool get_instance_setting_string(const KODI_ADDON_INSTANCE_BACKEND_HDL hdl,
+                                          const char* id,
+                                          char** value);
+  static bool set_instance_setting_bool(const KODI_ADDON_INSTANCE_BACKEND_HDL hdl,
+                                        const char* id,
+                                        bool value);
+  static bool set_instance_setting_int(const KODI_ADDON_INSTANCE_BACKEND_HDL hdl,
+                                       const char* id,
+                                       int value);
+  static bool set_instance_setting_float(const KODI_ADDON_INSTANCE_BACKEND_HDL hdl,
+                                         const char* id,
+                                         float value);
+  static bool set_instance_setting_string(const KODI_ADDON_INSTANCE_BACKEND_HDL hdl,
+                                          const char* id,
+                                          const char* value);
+
+  const ADDON_TYPE m_type;
+  const uint32_t m_instanceNumber;
```

Comment:
initialize the member here. maybe also others

Reply:
Is it needed, as all becomes set by class constructor?

Reply:
Also here I added the `const` on begin and without set in constructor it can never works.

Reply:
> Is it needed, as all becomes set by class constructor?

 Remove it from the ctor. Benefit of init here is once you have more than one ctor and it is clear from reading the header hoe this is initialized.